### DISCUSSION TITLE
Determine if data request is loader or action

### DISCUSSION THREAD

**Discussion by liegeandlief** [remix-run/remix#2272] [Q&A] [answered]
Is there a way that I can determine in the `handleDataRequest` function whether the data request is a loader or action request?

I could check the request method but this seems a bit brittle. It would be helpful if a parameter was passed into `handleDataRequest` called something like `type` with the value `loader` or `action`.

**Comment by bk-tho** [comment, marked as answer]
> I could check the request method but this seems a bit brittle

I don't think that's brittle at all. `GET` => `loader`, anything else => `action`. Highly unlikely that this will change in the future.

**Comment by liegeandlief** [comment]
@bk-tho are `HEAD` requests not also handled by the loader?

**Reply by sergiodxa** [reply]
Yes, HEAD is also sent to the loader.
